backup/src/main/java/com/kin/ecosystem/backup/qr/QRFileUriHandlerImpl.java

```diff
@@ -0,0 +1,81 @@
+package com.kin.ecosystem.backup.qr;
+
+
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.ParcelFileDescriptor;
+import android.support.annotation.NonNull;
+import android.support.v4.content.FileProvider;
+import java.io.File;
+import java.io.FileDescriptor;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+public class QRFileUriHandlerImpl implements QRFileUriHandler {
+
+	private static final String RELATIVE_PATH_FILENAME_QR_IMAGE = "/qr_codes/backup_qr.png";
+	private final Context context;
+
+	public QRFileUriHandlerImpl(@NonNull Context context) {
+		this.context = context;
+	}
+
+	@NonNull
+	@Override
+	public Bitmap loadFile(@NonNull Uri uri) throws IOException {
+		ParcelFileDescriptor pfd = context.getContentResolver().openFileDescriptor(uri, "r");
+		if (pfd != null) {
+			FileDescriptor fd = pfd.getFileDescriptor();
+			Bitmap bitmap = BitmapFactory.decodeFileDescriptor(fd);
+			if (bitmap != null) {
+				return bitmap;
+			}
+		}
+		throw new IOException("decoding file as bitmap failed.");
+	}
+
+	@NonNull
+	@Override
+	public Uri saveFile(@NonNull Bitmap bitmap) throws IOException {
+		File file = getOrCreateSaveFile();
+		FileOutputStream stream = new FileOutputStream(file);
+		bitmap.compress(Bitmap.CompressFormat.PNG, 90, stream);
```

Comment:
do we ever delete this file?

Reply:
we can't do it reliably, meaning we don't know when the file can be deleted.